knora/dsplib/utils/stack_handling.py

```diff
@@ -0,0 +1,122 @@
+            if response.ok:
+                break
+        except:
+            time.sleep(1)
+        time.sleep(1)
+
+    # inside fuseki, create the "knora-test" repository
+    repo_template = requests.get(f"{url_prefix}webapi/scripts/fuseki-repository-config.ttl.template").text
+    repo_template = repo_template.replace("@REPOSITORY@", "knora-test")
+    response = requests.post(
+        url="http://0.0.0.0:3030/$/datasets",
+        files={"file": ("file.ttl", repo_template, "text/turtle; charset=utf8")},
+        auth=("admin", "test")
+    )
+    if not response.ok:
+        raise BaseError("Cannot start DSP-API: Error when creating the 'knora-test' repository. Is DSP-API perhaps already running? "
+                        "another DSP-API running already?")
+
+    # load some basic ontos and data into the repository
+    graph_prefix = "http://0.0.0.0:3030/knora-test/data?graph="
+    ttl_files = [
+        ("knora-ontologies/knora-admin.ttl", "http://www.knora.org/ontology/knora-admin"),
+        ("knora-ontologies/knora-base.ttl", "http://www.knora.org/ontology/knora-base"),
+        ("knora-ontologies/standoff-onto.ttl", "http://www.knora.org/ontology/standoff"),
+        ("knora-ontologies/standoff-data.ttl", "http://www.knora.org/data/standoff"),
+        ("knora-ontologies/salsah-gui.ttl", "http://www.knora.org/ontology/salsah-gui"),
+        ("test_data/all_data/admin-data-minimal.ttl", "http://www.knora.org/data/admin"),
+        ("test_data/all_data/permissions-data-minimal.ttl", "http://www.knora.org/data/permissions")
+    ]
+    for ttl_file, graph in ttl_files:
+        ttl_text = requests.get(url_prefix + ttl_file).text
+        response = requests.post(
+            url=graph_prefix + graph,
+            files={"file": ("file.ttl", ttl_text, "text/turtle; charset: utf-8")},
+            auth=("admin", "test"),
+        )
+        if not response.ok:
+            raise BaseError(f"Cannot start DSP-API: Error when creating graph '{graph}'")
+
+    # startup all other components
+    subprocess.run("docker compose up -d", shell=True, cwd=docker_path)
+    print("DSP-API is now running on http://localhost:3333/ and DSP-APP on http://localhost:4200/")
+
+    # docker system prune
+    if enforce_docker_system_prune:
+        prune_docker = "y"
+    elif suppress_docker_system_prune:
+        prune_docker = "N"
+    else:
+        prune_docker = None
+        while prune_docker not in ["y", "N"]:
+            prune_docker = subprocess.run(
+                "read -p \"Allow dsp-tools to execute a docker system prune? This is necessary to keep your Docker clean. If "
+                "you are unsure what that means, just type y and press Enter. [y/N]\" response; echo $response",
+                shell=True,
+                stdout=subprocess.PIPE,
+                text=True
+            ).stdout.replace("\n", "")
+    if prune_docker == "y":
+        subprocess.run("docker system prune -f", shell=True, cwd=docker_path)
```

Comment:
this seems overly complicated to me: why delegate the asking to the docker process? you could simply prompt the user yourself wether or not they want "prune" (in a separate function that returns `true` if they typed "y" or "Y", `false` if "n" or "N", and takes care of re-asking in any other case, maybe `_do_prune()`) and then do `if enforce_docker_system_prune or _do_prune(): ...`

But more generally: Wouldn't it be simpler to define a default behaviour (either prune or no prune, that's up to you), and then only have a flag for the other case.

Reply:
fair enough - and I agree that pruning by default would be risky.  
Then I still think it would be simpler in code and more user friendly not to have this "ask" case, where you pipe stuff back and forth between DSP-Tools and docker, but rather you ask them `input("do you want to call 'docker prune'? [Y/N ] ")`

Reply:
oops... writing at the same time... and now I understand what's happening :).
yes, that will streamline it a bit.